File: Data Structures & Algorithms/top-k-elements-in-list/submission-2.cpp
class Solution {
public:
    
    vector<int> topKFrequent(vector<int>& nums, int k) {
        map<int, int> buffer;
        for (auto num: nums){
            buffer[num]++;
        }

        vector<pair<int, int>> counts;
        auto it = buffer.begin();
        while(it != buffer.end()){
            counts.push_back({it->second, it->first});
            it++;
        }

        sort(counts.rbegin(), counts.rend());
        vector<int> result;
        int idx = 0;
        while(k > 0){
            result.push_back(counts[idx].second);
            idx++;
            k--;
        }
        return result;
    }
};

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-2.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        result = [0]*len(nums)
        zero_count = 0
        product = 1
        for i in range(len(nums)):
            if nums[i] == 0:
                zero_count += 1
            else:
                product *= nums[i]

        if zero_count > 1:
            return result
        
        for j in range(len(nums)):
            if zero_count:
                result[j] = product if nums[j] == 0 else 0
            else:
                result[j] = int(product/nums[j])
        return result



